src/rgw/rgw_op.cc

```diff
@@ -6239,7 +6242,7 @@ bool RGWBulkDelete::Deleter::verify_permission(RGWBucketInfo& binfo,
 
   /* We can use global user_acl because each BulkDelete request is allowed
    * to work on entities from a single account only. */
-  return verify_bucket_permission(s, binfo.bucket, s->user_acl.get(),
+  return verify_bucket_permission(s, s, binfo.bucket, s->user_acl.get(),
```

Comment:
@cbodley In this `verify_bucket_permission` I couldn't pass `this` because `RGWBulkDelete::Deleter` isn't an RGWOp. `Deleter` is a class declared in `RGWBulkDelete` which is an `RGWOp`. Would you do something differently?

Reply:
i see what you mean. maybe Deleter's constructor could take the dpp as an argument.. something like this?
```c++
void RGWBulkDelete::execute()
{
  deleter = std::unique_ptr<Deleter>(new Deleter(this, store, s));
```